packages/grafana-data/src/transformations/transformers/convertFieldType.test.ts

```diff
@@ -183,6 +184,87 @@ describe('field convert types transformer', () => {
     ]);
   });
 
+  it('will convert JSON fields to complex objects', () => {
+    const options = {
+      conversions: [
+        { targetField: 'numbers', destinationType: FieldType.other, inputFormat: ComplexFieldParserID.JSON },
+        { targetField: 'objects', destinationType: FieldType.other, inputFormat: ComplexFieldParserID.JSON },
+        { targetField: 'arrays', destinationType: FieldType.other, inputFormat: ComplexFieldParserID.JSON },
+        { targetField: 'invalids', destinationType: FieldType.other, inputFormat: ComplexFieldParserID.JSON },
+        { targetField: 'mixed', destinationType: FieldType.other, inputFormat: ComplexFieldParserID.JSON },
+      ],
+    };
+
+    const comboTypes = toDataFrame({
+      fields: [
+        {
+          name: 'numbers',
+          type: FieldType.number,
+          values: [-1, 1, null],
+        },
+        {
+          name: 'objects',
+          type: FieldType.string,
+          values: [
+            '{ "neg": -100, "zero": 0, "pos": 1, "null": null, "array": [0, 1, 2], "nested": { "number": 1 } }',
+            '{ "neg": -10, "zero": 0, "pos": 2, "null": null, "array": [2, 3, 4], "nested": { "number": 2 } }',
+            '{ "neg": -1, "zero": 0, "pos": 3, "null": null, "array": [3, 4, 5], "nested": { "number": 3 } }',
```

Comment:
```suggestion
            '[1,2,3,4]',
            'non json value', // error or empy?
```

can the test exercise different shapes rather than same shape with different values?